### PR TITLE
feat: Support More Endpoints

### LINKED ISSUE #378: Add in new `states` Interface service.
https://aws.amazon.com/about-aws/whats-new/2019/12/aws-step-functions-now-supports-aws-privatelink/

### PR DESCRIPTION
**Pull request by DrFaust92**
# Description
Support Additional VPC Endpoints:

Closes #378

- Data Sync
- EBS
- Elastic Inference Runtime
- SMS
- Access Analyzer
- Workspaces
- Auto Scaling Plans
- QLDB Session
- Step Function
- Elastic Beanstalk + Health
- ACM PCA
- EMR

### CODE REVIEW

**Comment by DrFaust92** [comment]
@antonbabenko, i'm thinking whether the variables file became too bloated because of endpoint vars. maybe split the endpoint vars to another file (like the endpoint resources)? same question for output.

If its split i thought it can be easier to maybe generate these files based on a closed list of endpoints, adding new endpoints will be easier that way.

**Comment by DrFaust92** [comment]
@antonbabenko , can you take a look at this one?

File: outputs.tf [hunk 1]
```diff
@@ -1017,6 +1017,201 @@ output "vpc_endpoint_cloud_directory_dns_entry" {
...
+}
+
+output "vpc_endpoint_access_analyzer_dns_entry" {
+  description = "The DNS entries for the VPC Endpoint for Access Analyzer."
+  value       = flatten(aws_vpc_endpoint.access_analyzer.*.dns_entry)
+}
+
+output "vpc_endpoint_acm_pca_id" {
+  description = "The ID of VPC endpoint for ACM PCA"
+  value       = concat(aws_vpc_endpoint.access_analyzer.*.id, [""])[0]
```
Comment (danquack) [inline outputs.tf:1202]: This value seems like a copy pasta issue

**Comment by antonbabenko** [comment]
Thanks @DrFaust92 for your work (as usual)!

[v2.36.0](https://github.com/terraform-aws-modules/terraform-aws-vpc/releases/v2.36.0) has been just released.

**Pull request merged**